### PR TITLE
Add gawk

### PR DESCRIPTION
**Pull request by bollwyvl**
😊 haven't tried the windows build locally, but here's hoping!

### CODE REVIEW

**Comment by bollwyvl** [comment]
Actually got the windows build working, but `make install` is sad because it can't find `cat`... because the install is written _in_ `awk`, this suggests it might be a runtime dependency... still investigating...

**Comment by jakirkham** [comment]
I think you want `m2-coreutils` if you want `cat`.

**Comment by bollwyvl** [comment]
> I think you want `m2-coreutils` if you want `cat`.

Indeed, that did it! Though it does _not_ appear to be used at runtime

I couldn't get `make check` working just yet on windows, but otherwise am getting decent builds there. Here's hoping!

**Comment by bollwyvl** [comment]
Running too many windows builds, trying with pinned python...

**Comment by bollwyvl** [comment]
Hm... seems all the win64 builds are hanging on `cat`...

**Comment by bollwyvl** [comment]
So the `m2-patch` was an artifact of how i set up my local vagrant box. Everything seems peachy now!

File: recipes/gawk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - win.gawkmisc.patch  # [win]
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - toolchain
+    - m2w64-toolchain  # [win]
+    - python  # [win]
```
Comment (bollwyvl) [inline recipes/gawk/meta.yaml]: Of course, it would be better if I had awk around to rewrite my awk... or even sed...

**Comment by bollwyvl** [comment]
Yeah... giving up on getting `make check` to run in windows. Otherwise, pretty much ready to let this ride...

**Comment by jakirkham** [comment]
I think we need the `vc` features logic for Windows still.

**Comment by jakirkham** [comment]
Or do we if we are using MinGW? @mingwandroid could you please help us here?

**Comment by mingwandroid** [comment]
The problem with building `gawk` on Windows with the `mingw-w64` toolchain is that it will not handle CRLF correctly (or be able to fork - don't know if gawk does that though - or do path conversions when it is fed an msys2 path via a file).

Instead, for Windows, you should just make this a metapackage that depends upon `m2-gawk`.

**Comment by bollwyvl** [comment]
Cool, will look into that!

**Comment by bollwyvl** [comment]
Ah, i see:
- as with `cat`, I just hadn't perused enough what was available in `m2-`
- the end-goal package, which requires `gawk`, doesn't even really support a mingw build on windows anyway, and even then is `sln` based

So I am inclined to `skip` windows on this recipe, and the goal package, until i have a need for it, though I do think having a conda-forge `gawk` is a fine thing to do. I'll skip appropriately.

**Comment by mingwandroid** [comment]
The end goal package probably wants to run gawk during build? In that case (and anyway) `m2-gawk` will likely just work. Interop is a guiding principle with `m2-*`

**Comment by bollwyvl** [comment]
> The end goal package probably wants to run gawk during build? 

Yeah, presumably at some point, but [it starts with](https://github.com/openlink/virtuoso-opensource/blob/develop/7/README.WINDOWS) "open up your VS 2003," so I'm not going to bother at this point.

On unix, I also dropped all the readline, mpfr stuff which, while no doubt useful, would probably need cb3 variants to work robustly.

File: recipes/gawk/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win]
+  script:
+    - ./configure --prefix="{{ PREFIX }}"
+    - make -j${NUM_CPUS}
+    - make check -j${NUM_CPUS}
+    - make install
```
Comment (jakirkham) [inline recipes/gawk/meta.yaml]: Normally would put this in `build.sh`, but it is pretty short.

**Comment by jakirkham** [comment]
Looks great. Thanks @bollwyvl. Also thanks @mingwandroid for the assist.

**Pull request merged**